frontend/src/store/modules/github.ts

```diff
@@ -0,0 +1,83 @@
+import axios from "axios";
+import {
+  ExternalRepositoryInfo,
+  VCS,
+  OAuthConfig,
+  OAuthToken,
+} from "../../types";
+
+const getters = {};
+
+function convertGitHubProject(project: any): ExternalRepositoryInfo {
+  return {
+    externalId: project.id.toString(),
+    name: project.name,
+    fullPath: project.path_with_namespace,
+    webUrl: project.web_url,
+  };
+}
+
+const actions = {
+  async exchangeToken(
+    {}: any,
+    {
+      oAuthConfig,
+      code,
+    }: {
+      oAuthConfig: OAuthConfig;
+      code: string;
+    }
+  ): Promise<OAuthToken> {
+    const data = (
+      await axios.post(
+        // FIXME: This is damn unsafe, only OK with development, a real solution must use backend to make reqeusts to the GitHub.com
+        `https://cors-anywhere.herokuapp.com/${oAuthConfig.endpoint}?client_id=${oAuthConfig.applicationId}&client_secret=${oAuthConfig.secret}&code=${code}&redirect_uri=${oAuthConfig.redirectUrl}&grant_type=authorization_code`
```

Comment:
TIL

Reply:
@zhouzilong2020 is working on to move the GitLab implementation to the backend.

I am thinking of having a POST /oauth/exchange-token endpoint.